src/sg/flybot/pullable/pattern.clj

```diff
@@ -70,5 +73,79 @@
        (pattern-error! "unable to understand" pattern)))))
 
 (comment
-  (->query #(concat % ['ok]) '{(:a :with [{:b 2 :c 3}]) {:b ?}})
+  (->query #(concat % ['ok]) '{(:a :with [{:b 2 :c 3}]) {:b ?}}) 
   )
+
+(defn valid-symbol?
+  "Returns true if `p` is a symbol starting with ?"
+  [p]
+  (and (symbol? p) (re-matches #"\?.*" (name p))))
+
+
+(def sch-pattern-proc
+  "Malli schema for the different post-processors in the keys.
+   Expects individual post-processors with format [proc arg]
+   i.e. [:not-found 0]"
+  [:multi {:dispatch first}
+   [:when [:tuple [:= :when] fn?]]
+   [:not-found [:tuple [:= :not-found] :any]]
+   [:with [:tuple [:= :with] vector?]]
+   [:batch [:tuple [:= :batch] [:vector vector?]]]
+   [:watch [:tuple [:= :watch] fn?]]])
+
+(defn valid-proc-keys?
+  "Validates the key with post-processors such as (:a :not-found 0 :when even?)."
+  [k] 
+  (->> k
+       rest
+       (partition 2) 
+       (map #(m/validate sch-pattern-proc (into [] %))) 
```

Comment:
`validate` need to reconstruct a validator function every time, because the schema seldom changes, remember the validator function might be more efficient.

Reply:
I now transform the `keys` of the pattern of the pattern into vector of vectors **before** the validation.

So there is no intermediate `m/validate` for the keys anymore